insights/client/client.py

```diff
@@ -442,8 +446,9 @@ def upload(config, pconn, tar_file, content_type, collection_duration=None):
     if config.legacy_upload:
         return _legacy_upload(config, pconn, tar_file, content_type, collection_duration)
     logger.info('Uploading Insights data.')
-
+    pid = read_pidfile()
```

Comment:
also here (see comment below (or above? I'm not sure what order this will appear))